server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1409,6 +1410,12 @@ private Pair<List<Long>, Integer> searchForUserVMIdsAndCount(ListVMsCmd cmd) {
         }
 
         Boolean isVnf = cmd.getVnf();
+        if (cmd instanceof ListVnfAppliancesCmd) {
+            if (isVnf != null && !isVnf) {
+                throw new InvalidParameterValueException("Unable to list VNF appliances as isvnf is set to false");
+            }
+            isVnf = true;
+        }
```

Comment:
shouldn't the command class set the falg to true in it's constructor and not allow it to be changed?

Reply:
if `isVnf` is set to false, should throw an exception ?

Reply:
well, if we get the listVnfAppliences API call we shouldn't bother about the flag at all, I think. The API implies the flag is true.

Reply:
pr is updated. thanks @DaanHoogland